src/server/wesnothd/server.cpp

```diff
@@ -2131,6 +2134,49 @@ void server::requests_handler(const std::string& /*issuer_name*/,
 	metrics_.requests(*out);
 }
 
+void server::roll_handler(const std::string& issuer_name,
+		const std::string& /*query*/,
+		std::string& parameters,
+		std::ostringstream* out)
+{
+	assert(out != nullptr);
+	if(parameters.empty()) {
+		return;
+	}
+	
+	int N;
+	try {
+		N = std::stoi(parameters);
+	} catch(const std::invalid_argument&) {
+		*out << "The number of die sides must be a number!";
+		return;
+	} catch(const std::out_of_range&) {
+		*out << "The number of sides is too big for the die!";
+		return;
+	}
+
+	if(N < 1) {
+		*out << "The die cannot have less than 1 side!";
+		return;
+	}
+	std::uniform_int_distribution<int> dice_distro(1, N);
+	std::string value = std::to_string(dice_distro(die_));
+
+	*out << "You rolled a die [1 - " + parameters + "] and got " + value + ".";
+
+	auto player_ptr = player_connections_.get<name_t>().find(issuer_name);
+	if(player_ptr == player_connections_.get<name_t>().end()) {
+		return;
+	}
+
+	auto g_ptr = player_ptr->get_game();
+	if(g_ptr && g_ptr->is_member(player_ptr->socket())) {
+		g_ptr->send_server_message_to_all(issuer_name + " rolled a die [1 - " + parameters + "] and got " + value + ".", player_ptr->socket());
+	} else {
+		*out << " (The result is shown to the others only in game lobbies.)";
```

Comment:
This is not entirely accurate since it's also shown once the game is started, right?

Perhaps should be: " (The result is shown to others only in a game.)"